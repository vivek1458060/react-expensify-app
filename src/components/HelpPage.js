import React from 'react';

const HelpPage = () => (
    <div>
        <p>This is my Help Page!</p>
    </div>
);

export default HelpPage;