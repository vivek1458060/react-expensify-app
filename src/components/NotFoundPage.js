import React from 'react';
import {Link} from 'react-router-dom';

export const NotFoundPage = () => (
    <div>
        <p>404 - <Link to="/">Go Home</Link></p>
    </div>
);

export default NotFoundPage;