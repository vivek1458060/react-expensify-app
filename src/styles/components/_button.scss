.button {
    background: $blue;
    border:none;
    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;
    padding: $s-size;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}