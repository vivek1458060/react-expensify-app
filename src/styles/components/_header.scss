.header {
    background: $dark-blue;
}

.header__content {
    display: flex;
    justify-content: space-between;
    padding: $s-size;
    align-items: center;
}

.header__title {
    color: white;
    text-decoration: none;
    h1 {
        margin: 0;
    }
}