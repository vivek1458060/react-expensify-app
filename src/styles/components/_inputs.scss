.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}