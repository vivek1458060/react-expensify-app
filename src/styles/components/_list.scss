.list-header {
    background: $off-white;
    color: $grey;
    border: 1px solid darken($off-white, 7%);
    padding: $s-size $m-size;
    display: flex;
    justify-content: space-between;
}

.list-body {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

.list-item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    text-decoration: none;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    @media (min-width: $desktop-breakpoint){
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $grey;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title {
    color: $grey;
    font-size: $font-size-small;
}

.list-item__data {
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}