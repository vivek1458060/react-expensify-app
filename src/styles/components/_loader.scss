.loader {
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

.loader__image {
    height: 6rem;
    width: 6rem;
}