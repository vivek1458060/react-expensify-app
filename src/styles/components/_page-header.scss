.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page-header__title {
    margin: 0;
    font-weight: 300;
    span {
        font-weight: 700;
    }
}

.page-header__actions {
    margin-top: $m-size;
}