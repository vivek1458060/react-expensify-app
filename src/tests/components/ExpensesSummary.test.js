import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should correctly render ExpenseSummary with on expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={2355}/>);
    expect(wrapper).toMatchSnapshot();
});