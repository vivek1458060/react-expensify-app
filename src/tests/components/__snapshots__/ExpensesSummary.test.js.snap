// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $23.55
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with on expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;
